### PR TITLE
Fixes VM Power State Issue

### LINKED ISSUE #6: Enforced VM Power State
The current implement of the xVMHyper-V resource enforces that a VM is in a particular power state. If no state is specified then it's powered off!

I think that this is incorrect behaviour as it will turn off VMs __unless__ you explicitly state you want VMs running __all__ the time. In a lab/dev/test (or maybe even a production) environment, you might want to ensure that VMs are provisioned in a particular way, but managing the power state is not always desirable (pun intended!).

I think this should be the behaviour:

* If the configuration document specifies a State of 'Off', 'Running' or 'Paused' then this should be enforced.
* If the configuration document does not specify a State, then no change to the VM should be made.

### PR DESCRIPTION
**Pull request by iainbrighton**
* FIxes #6
* Removed default 'Off' setting from the xVMHyper-V $State property
* Added check for $State property being set before calling Set-VMState

### CODE REVIEW

File: DSCResources/MSFT_xVMHyperV/MSFT_xVMHyperV.psm1 [hunk 1]
```diff
@@ -136,8 +137,8 @@ function Set-TargetResource
         # One cannot set the VM's vhdpath, path, generation and switchName after creation 
         else
         {
-            # If the VM is not in right state, set it to right state
-            if($vmObj.State -ne $State)
+            # If state has been specified and the VM is not in right state, set it to right state
+            if($State -and $vmObj.State -ne $State)
```
Comment (vors) [inline DSCResources/MSFT_xVMHyperV/MSFT_xVMHyperV.psm1]: Nit: I would prefer `($State -and ($vmObj.State -ne $State))` for readability

**Comment by vors** [comment]
Hi @iainbrighton! Thank you for a great suggestion in #6 and fix that implement it. We are trying to increase quality bar and test coverage. We encouraging contributor cover all behavior changes with pester tests. Can you add tests for your changes? Let me know if you would have any problems with it.

**Comment by iainbrighton** [comment]
Of course! Do you have a template/example file that you would like me to use or leave it to artistic license?

**Comment by vors** [comment]
[xSharePoint tests](https://github.com/PowerShell/xSharePoint/tree/master/Tests) are pretty good, please them as an example. JFYI, We are discussion contribution guidelines for tests [here](https://github.com/PowerShell/DscResources/issues/9).

**Comment by iainbrighton** [comment]
Thanks for the pointers! I've started work on the unit tests and will hopefully finish them after a busy weekend :)

**Pull request closed without merging**